lib/experimental/block-editor-settings.php

```diff
@@ -5,6 +5,30 @@
  * @package gutenberg
  */
 
+/**
+ * Finds the first occurrence of a specific block in an array of blocks.
+ *
+ * @param string $block_name Name of the block to find.
+ * @param array  $blocks     Array of blocks.
+ * @return array Found block, or empty array if none found.
+ */
+function gutenberg_find_first_block( $block_name, $blocks ) {
+	foreach ( $blocks as $block ) {
+		if ( $block_name === $block['blockName'] ) {
+			return $block;
+		}
+		if ( ! empty( $block['innerBlocks'] ) ) {
+			$post_content = gutenberg_find_first_block( $block_name, $block['innerBlocks'] );
```

Comment:
Tiny nit: since this function has been abstracted away from being solely used for the post content block, would it be better to name this variable `$found_block` rather than `$post_content`?

Reply:
@andrewserong Good catch, updated!